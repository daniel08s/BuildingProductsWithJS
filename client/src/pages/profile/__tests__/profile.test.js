/* global test expect shallow React */
// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import ProfileWrapper, {Profile} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '1',
  login: 'testuser',
  registrationDate: new Date(2018, 1, 1, 1, 1, 1, 1),
};

test('# Profile page wrapper', () => {
  const store = mockStore({auth: {user}, users: {user}});

  const wrapper = shallow(<ProfileWrapper store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Profile page', () => {
  const getUser = (u) => {
    expect(u).toEqual(user);
  };
  const component = (
    <Profile
      user={user}
      loadedUser={user}
      getUser={getUser}
      params={user}
    />
  );

  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});
