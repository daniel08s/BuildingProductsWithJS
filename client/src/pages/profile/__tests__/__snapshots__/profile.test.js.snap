// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Profile page 1`] = `
<div
  className="container"
>
  <Component
    current="/profile/1"
    user={
      Object {
        "id": "1",
        "login": "testuser",
        "registrationDate": 2018-02-01T01:01:01.001Z,
      }
    }
  />
  <Connect(User)
    edit={true}
    user={
      Object {
        "id": "1",
        "login": "testuser",
        "registrationDate": 2018-02-01T01:01:01.001Z,
      }
    }
  />
</div>
`;

exports[`# Profile page wrapper 1`] = `
<Profile
  getUser={[Function]}
  loadedUser={
    Object {
      "id": "1",
      "login": "testuser",
      "registrationDate": 2018-02-01T01:01:01.001Z,
    }
  }
  params={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  user={
    Object {
      "id": "1",
      "login": "testuser",
      "registrationDate": 2018-02-01T01:01:01.001Z,
    }
  }
/>
`;
